src/main/java/org/bookstore/controller/BookController.java

```diff
@@ -29,7 +32,18 @@ public BookDto getBookById(@PathVariable Long id) {
     }
 
     @PostMapping
-    public BookDto createBook(@RequestBody CreateBookRequestDto bookRequestDto) {
+    public BookDto createBook(@RequestBody @Valid CreateBookRequestDto bookRequestDto) {
         return bookService.save(bookRequestDto);
     }
+
+    @PutMapping("/{id}")
+    public void updateBookById(@PathVariable Long id,
+                               @RequestBody @Valid CreateBookRequestDto bookRequestDto) {
+        bookService.updateBookById(id, bookRequestDto);
+    }
+
+    @DeleteMapping("/{id}")
+    public void deleteBookById(@PathVariable Long id) {
```

Comment:
just FYI

you may also validate such things as @PathVariable, @RequestParams etc. It looks like:
```
@PathVariable @Positive Long id
```

In order to be able to perform such validation you need to annotate your controller class with `@Validated` annotation

again, this is just FYI - apply it or not :)